src/snowflake/snowpark/modin/plugin/extensions/index.py

```diff
@@ -92,17 +150,34 @@ def __init__(
         >>> pd.Index([1, 2, 3], dtype="uint8")
         Index([1, 2, 3], dtype='int64')
         """
+        kwargs = {
+            "dtype": dtype,
+            "copy": copy,
+            "name": name,
+            "tupleize_cols": tupleize_cols,
+        }
+        self._init_index(data, _CONSTRUCTOR_DEFAULTS, **kwargs)
+
+    def _init_index(
+        self,
+        data: ArrayLike | SnowflakeQueryCompiler | None,
+        ctor_defaults: dict,
+        **kwargs: Any,
+    ):
+        if isinstance(data, SnowflakeQueryCompiler):
+            # Raise warning if `data` is query compiler with non-default arguments.
+            for arg_name, arg_value in kwargs.items():
+                if arg_value is not ctor_defaults[arg_name]:
+                    WarningMessage.ignored_argument(
```

Comment:
I think we should raise assertion error since qc data is only used by us not the user.

Reply:
Added assert